api/src/domain/user/encryptTaxIdBatch.test.ts

```diff
@@ -0,0 +1,69 @@
+import { UserData } from "@shared/userData";
+import { generateProfileData, generateUserData } from "../../../test/factories";
+import { EncryptionDecryptionClient, EncryptTaxId, UserDataClient, UserDataQlClient } from "../types";
+import { encryptTaxIdBatch } from "./encryptTaxIdBatch";
+import { encryptTaxIdFactory } from "./encryptTaxIdFactory";
+
+describe("encryptTaxIdBatch", () => {
+  let stubEncryptionDecryptionClient: jest.Mocked<EncryptionDecryptionClient>;
+  let encryptTaxId: EncryptTaxId;
+  let stubUserDataClient: jest.Mocked<UserDataClient>;
+  let stubUserDataQlClient: jest.Mocked<UserDataQlClient>;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    stubEncryptionDecryptionClient = { encryptValue: jest.fn(), decryptValue: jest.fn() };
+    stubUserDataClient = {
+      get: jest.fn(),
+      put: jest.fn(),
+      findByEmail: jest.fn(),
+    };
+    stubUserDataQlClient = {
+      getNeedNewsletterUsers: jest.fn(),
+      getNeedToAddToUserTestingUsers: jest.fn(),
+      getNeedTaxIdEncryptionUsers: jest.fn(),
+      search: jest.fn(),
+    };
+    encryptTaxId = encryptTaxIdFactory(stubEncryptionDecryptionClient);
+  });
+
+  it("adds all users and returns success, failed, and total count when all succeed", async () => {
+    stubUserDataClient.put.mockImplementation((userData: UserData): Promise<UserData> => {
+      return Promise.resolve(userData);
+    });
```

Comment:
this works but could be significantly simplified if you used `mockResolvedValue`